pkg/knowledge_base2/edge_template.go

```diff
@@ -12,6 +12,8 @@ type (
 		Source construct.ResourceId `yaml:"source"`
 		Target construct.ResourceId `yaml:"target"`
 
+		AlwaysProcess bool `yaml:"always_process"`
```

Comment:
Can you godoc this? What does it mean?

I understand its purpose, but couldn't we determine whether it needs to be recomputed based on dependencies that changed? Is this just a temporary solution for that?

Reply:
we never add edges to the operational view so they never get recomputed if they already exist. This is an existing bug for deployment triggers i realized and also happened when doing cdn behaviors. Ill add a comment, but seems like the amount of work to approach this in a programmatic manner is really high

Reply:
Because this would mean that if a user added their own custom behaviour, that it would get overwritten the next time the engine ran, right?

Reply:
no since our configuration rules only ever do append it should still be there

Reply:
Would this make it impossible to only expose a part of a Rest API via the cloudfront distribution?

Reply:
currently yes, if the api is not the default origin. if we want cdn to have a edge to the integration thats another route, but i dont think thats as intuitive.

Reply:
I think if it's CDN -> Rest API, then it should just have a `ANY /*` behaviour to indicate that it's pass-through. If it's per-route, then it should be CDN -> Integration corresponding with that specific route. This would be the most intuitive to me, and the most flexible as it would be capable of both forms.

Reply:
i disagree otherwise you cant have multiple origins

Reply:
by doing the integrations its at least allowing multiple origins as long as their paths dont conflict. This is the ux aaron and i had discussed so if we want to change it we can bring it up in the channel